Netatmo.lua

```diff
@@ -307,15 +318,33 @@ function QuickApp:getNetatmoDevicesData(token, mode)
     )
 end
 
+function QuickApp:addInterface(id, param)
+    local device = api.get('/devices/' .. tostring(id))
+    for _, interface in ipairs(device.interfaces) do
+        if interface == param then
```

Comment:
This condition never met because child doesn't have "battery" interface already. Previously you had a function to check if device has an "battery" interface.

Reply:
Exact, I have modified this part of the code

`
function QuickApp:addInterface(child, param)
    local device = api.get('/devices/' .. tostring(child.id))
    local found = false
    for _, interface in ipairs(device.interfaces) do
        if interface == param then
            found = true
            break
        end
    end
    if not found then
        self:debug("Add '" .. param .. "' interface to device #" .. tostring(device.id))
        child:addInterfaces({param})
    end
end
`